### PR TITLE
aruha-148 make event type schema optional

### LINKED ISSUE #145: EntityType.owning_application is required?
Creation of events without `owning_application` fails with an error:

```
{
  "type": "http://httpstatus.es/422",
  "title": "Unprocessable Entity",
  "detail": "Field \"owning_application\" may not be null\n",
  "status": 422
}
```

Is `owning_application` a required field for `EventType` now? The API spec states otherwise.

```
{
 "name": "test",
 "category": "data",
 "schema": {
  "type": "json_schema",
  "schema": "{ \"properties\": { \"order_number\": { \"type\": \"string\" } } }"
 }
}
```

### PR DESCRIPTION
**Pull request by rcillo**
Schema should not be required, but if present then it should be
completely defined with schema.schema and schema.type.

This PR contains necessary adjustments to prevent null pointers.

Fixes #145

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/zalando/nakadi/pull/174" target="_blank">ReviewNinja

File: src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java [hunk 1]
```diff
@@ -148,18 +148,20 @@ public EventTypeController(final EventTypeRepository eventTypeRepository, final
...
-                throw new InvalidEventTypeException("\"metadata\" property is reserved");
-            } else {
-                SchemaLoader.load(schemaAsJson);
+        if (eventType.getSchema() != null) {
+            try {
+                final JSONObject schemaAsJson = new JSONObject(eventType.getSchema().getSchema());
+
+                if (hasReservedField(eventType, schemaAsJson, "metadata")) {
+                    throw new InvalidEventTypeException("\"metadata\" property is reserved");
+                } else {
```
Comment (jkliff) [inline src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java]: needless else ;)

Comment (rcillo) [inline src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java]: :+1: done https://github.com/zalando/nakadi/commit/b1eacbafa82ff99897ef9d99ac7823d548739e15

File: src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java [hunk 2]
```diff
@@ -183,7 +185,10 @@ private void validateName(final String name, final EventType eventType) throws I
     }
 
     private void validateSchemaChange(final EventType eventType, final EventType existingEventType) throws InvalidEventTypeException {
-        if (!existingEventType.getSchema().equals(eventType.getSchema())) {
+        boolean existingNonNullSchemaChanged = existingEventType.getSchema() != null && !existingEventType.getSchema().equals(eventType.getSchema());
```
Comment (CyberDem0n) [inline src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java]: final
final

Comment (rcillo) [inline src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java]: done :+1: https://github.com/zalando/nakadi/commit/7dadcb53e9e82c624979d1110d4b459289866450

**Comment by rcillo** [comment]
I've run the build again and it's actually green https://travis-ci.org/zalando/nakadi/builds/116122505

**Pull request merged**